packages/utils/lib/retry.ts

```diff
@@ -1,15 +1,16 @@
 interface RetryConfig {
     maxAttempts: number;
     delayMs: number | ((attempt: number) => number);
+    retryIf: (error: Error) => boolean;
 }
 
 export async function retry<T>(fn: () => T, config: RetryConfig): Promise<T> {
-    const { maxAttempts, delayMs } = config;
+    const { maxAttempts, delayMs, retryIf } = config;
     for (let attempt = 1; attempt <= maxAttempts; attempt++) {
         try {
             return fn();
         } catch (error) {
-            if (attempt < maxAttempts) {
+            if (attempt < maxAttempts && retryIf(error as Error)) {
                 const delay = typeof delayMs === 'number' ? delayMs : delayMs(attempt);
                 await new Promise((resolve) => setTimeout(resolve, delay));
```

Comment:
side note: we have access to timers/promise that allows us to do `await setTimeout(100)`

Reply:
no sure I follow

Reply:
just saying there is a better syntax but it's not part of this PR